core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -227,6 +227,7 @@ static void stop() {
      * This method is invoked by {@link Elasticsearch#main(String[])}
      * to startup elasticsearch.
      */
+    @SuppressForbidden(reason = "Allowed to exit explicitly in bootstrap phase")
```

Comment:
instead of suppressing the entire method can you wrap the forbidden calls in static methods in here and only suppress forbidden there? also applies to the other places :)